### ISSUE TITLE
NullReferenceException when accessing not existing section

### ISSUE THREAD

**Issue by rickyah** [rickyah/ini-parser#127]
POC
```csharp
IniData data = new IniData();

data["Section1"]["key1"] = "value1";

Console.WriteLine(data.ToString());
```
i got System.NullReferenceException in string : data["Section1"]["key1"] = "value1";

**Comment by Karaulov** [comment]
Yes!! Need fix it for remove unused code... (Adding sections is bad solution if many sections in many code, need rewrite all code to fix this issue, one of solutions - use another parser ?)

**Comment by caverna** [comment]
I'm not able to pull this commit, what do you think of this patch?
[0001-Quick-fix-for-127.zip](https://github.com/rickyah/ini-parser/files/690449/0001-Quick-fix-for-127.zip)

I believe that the default behaviour for INI is to return "" instead null.
And if a section doesn't exist, just create it! :-)

**Comment by caverna** [comment]
I got it! :-)

**Comment by Svetomech** [comment]
I don't see that you modified indexator anywhere. Was it not necessary?
Also, "" should be String.Empty, but that's a matter of preference.

**Comment by caverna** [comment]
String.Empty should be used, I agree...
I just wanted to create a quick fix... :-)
Look at line 92 and 94 in IniData.cs and line 86 in KeyDataCollection.
I just changed that...

**Comment by Svetomech** [comment]
Oh, silly me, found it. Github usually shows method's name above changes, but I guess this is not the case with indexators. Anyways, good job!

**Comment by Karaulov** [comment]
Any news?

**Comment by caverna** [comment]
@rickyah and I are discussing that

**Comment by rickyah** [comment]
Fixed. Latest code available on development branch, version 2.4 to be released before the end of the month
